Change to pip dependency install invocation during virtualenv setup in build

matt-y opened the pull request:
<!-- Please describe your changes on the following line: -->
Fix for https://github.com/servo/servo/issues/20990

In short, the pip invocation was causing my build on OSX to break. Using `python -m pip` in place of `pip` when setting up the virtual environment fixes the environment setup failure and allows the build to proceed. 

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #20990

<!-- Either: -->
- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/20998)
<!-- Reviewable:end -->

Commit:
Fix for pip invocation when setting up virtual environment during build

highfive commented:
Thanks for the pull request, and welcome! The Servo team is excited to review your changes, and you should hear from @avadacatavra (or someone else) soon.

highfive commented:
Heads up! This PR modifies the following files:
 * @wafflespeanut: python/mach_bootstrap.py

Commit:
removed sys.platform check & consolidated the pip invocations

The sys.platform check in need_pip_upgrade  was doing effectively the same pip invocation regardless of what
platform we were evaluating to.

Additionally, removed some duplicate definitions of the python variable which was already in scope
higher up in the function.

matt-y commented:
@jdm I've consolidated the pip invocations happening in the `if need_pip_upgrade:` block into a single `_process_exec` call as you suggested. 

I've also removed what appeared to be superfluous `if python` and the now-redundant `if pip` checks. `if python` is handled higher up in the function block.

jdm commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit cb18d8a has been approved by `jdm`

<!-- @bors-servo r=jdm cb18d8aff0bbd71db395339a5abfdb351ccf3d6d -->

bors-servo commented:
:hourglass: Testing commit cb18d8aff0bbd71db395339a5abfdb351ccf3d6d with merge 07ba57b864c00c820c601e420032d1ed682b6af2...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/4467)

tigercosmos commented:
@bors-servo retry

bors-servo commented:
:zap: Previous build results for [android](http://build.servo.org/builders/android/builds/11689), [arm32](http://build.servo.org/builders/arm32/builds/11913), [arm64](http://build.servo.org/builders/arm64/builds/11962), [linux-dev](http://build.servo.org/builders/linux-dev/builds/12003), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/8908), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/8911), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/12058), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/4487), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/8732), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/8785), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/4552), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/4485), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/7396) are reusable. Rebuilding only [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/4467)...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/11689), [arm32](http://build.servo.org/builders/arm32/builds/11913), [arm64](http://build.servo.org/builders/arm64/builds/11962), [linux-dev](http://build.servo.org/builders/linux-dev/builds/12003), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/8908), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/8911), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/12058), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/4469), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/4487), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/8732), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/8785), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/4552), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/4485), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/7396)
Approved by: jdm
Pushing 07ba57b864c00c820c601e420032d1ed682b6af2 to master...

tigercosmos commented:
@matt-y  Thanks
